diff --git a/bigtop-packages/src/common/kafka/do-component-build b/bigtop-packages/src/common/kafka/do-component-build
index e8c46187..9420b60a 100644
--- a/bigtop-packages/src/common/kafka/do-component-build
+++ b/bigtop-packages/src/common/kafka/do-component-build
@@ -21,6 +21,7 @@ set -ex
 BUILD_OPTS="-Divy.home=${HOME}/.ivy2 -Dsbt.ivy.home=${HOME}/.ivy2 -Duser.home=${HOME} \
             -Drepo.maven.org=$IVY_MIRROR_PROP \
             -Dreactor.repo=file://${HOME}/.m2/repository \
+            -DNEXUS_URL=${NEXUS_URL} \
             -DskipTests -DrecompileMode=all"

 # if BIGTOP_PREFERRED_JAVA_VERSION is set, switch java version for the build
@@ -36,7 +37,7 @@ export MAVEN_OPTS="-Xmx2g -XX:ReservedCodeCacheSize=512m"
 # Rewrite the zookeeper version defined in gradle/dependencies.gradle directly
 # with $ZOOKEEPER_VERSION, because we can't override it via command line option.
 sed -i -e 's/zookeeper: *"\([0-9]\{1,\}\.\)*[0-9]\{1,\}"/zookeeper: "'${ZOOKEEPER_VERSION}'"/' gradle/dependencies.gradle
-./gradlew -PscalaVersion=${SCALA_VERSION} clean releaseTarGz -x signArchives
+./gradlew -PscalaVersion=${SCALA_VERSION} clean releaseTarGz -x signArchives -d

 rm -rf build
 mkdir build
