diff --git a/ambari-admin/src/main/resources/ui/admin-web/.bowerrc b/ambari-admin/src/main/resources/ui/admin-web/.bowerrc
index d0f0b6fe31..797b71fcb7 100644
--- a/ambari-admin/src/main/resources/ui/admin-web/.bowerrc
+++ b/ambari-admin/src/main/resources/ui/admin-web/.bowerrc
@@ -1,3 +1,7 @@
 {
-    "directory": "app/bower_components"
+	"registry": "https://registry.bower.io/",
+	"strict-ssl": false,
+	"analytics": false,
+	"shorthand-resolver": "https://ghp.ci/https://github.com/{{owner}}/{{package}}.git",
+	"directory": "app/bower_components"
 }
\ No newline at end of file
